fastf1/core.py

```diff
@@ -1884,16 +1885,42 @@ def _check_lap_accuracy(self):
                                           atol=0.003, rtol=0, equal_nan=False)
                     if not check_2:
                         integrity_errors += 1
+                        lap_integrity_flag = False
                 else:
                     check_2 = False  # data not available means fail
 
                 if prev_lap is not None:
+                    if (not pd.isnull(prev_lap['Time'])):
+                        prev_lap_check = True  # it's needed for check_4
+                    else:
+                        prev_lap_check = False
                     # first lap after safety car often has timing issues (as do all laps under safety car)
                     check_3 = (prev_lap['TrackStatus'] != '4')
                 else:
+                    prev_lap_check = False
                     check_3 = True  # no previous lap, no SC error
 
-                result = check_1 and check_2 and check_3
+                pre_check_4 = (prev_lap_check
+                               & (not pd.isnull(lap['Time']))
+                               & (not pd.isnull(lap['LapTime'])))
+
+                if pre_check_4:  # needed condition for check_4
+                    time_diff = np.sum((lap['Time'], -1 * prev_lap['Time'])).total_seconds()
+                    lap_time = lap['LapTime'].total_seconds()
+                    # If the difference between the two times is within a
+                    # certain tolerance, the lap time data is considered to be valid.
+                    check_4 = np.allclose(time_diff, lap_time, atol=0.003, rtol=0, equal_nan=False)
+
+                    if not check_4 and lap_integrity_flag:
```

Comment:
The integrity flag is only used here and only set to False if ``not check_2``. Just use 
``if not check_2 and not check_4`` instead.

Reply:
In this way, if `check_2 is False`, `check_1 is True` and `check_4 is False` (I've still to catch if there is the possibility) it could happen to add two integrity errors for the same lap. Should we write:

`if not check_2 and not check_4 and check_1` instead?

Reply:
Ahh, I finally understand what you mean. I think a more readable and sustainable approach would be the following (although if this block of code gets any more complex in the future, this probably should be rewritten completely):

Remove your possible solution again. Introduce a ``lap_integrity_ok`` flag, that is set to ``True`` at the beginning of each iteration. Set ``lap_integrity_ok = False`` everywhere where there is currently a ``integrity_errors += 1`` statement.
Do a single ``if not lap_integrity_ok: integrity_errors += 1`` at the end of each iteration.
That removes the nested and difficult to understand if statements and it's shorter as well, I think.